### PR TITLE
fix(Ledger): Add Ledger Cardano app V5 voting support

### PR DESCRIPTION
**Pull request by michalszorad**
Resolves YOMO-615

TODO: 
- [x] Use yoroi-lib new version instead of local version. This needs to be updated in package.json

### CODE REVIEW

File: apps/wallet-mobile/src/components/ConfirmTx/ConfirmTx.tsx [hunk 1]
```diff
@@ -92,16 +99,17 @@ export const ConfirmTx = ({
...
     if (!wallet.hwDeviceInfo) throw new Error('No device info')
     setUseUSB(useUSB)
-    if (
-      (useUSB && wallet.hwDeviceInfo.hwFeatures.deviceObj == null) ||
-      (!useUSB && wallet.hwDeviceInfo.hwFeatures.deviceId == null)
-    ) {
+
+    const hwFeatures = wallet.hwDeviceInfo.hwFeatures
+
+    if ((useUSB && hwFeatures.deviceObj == null) || (!useUSB && hwFeatures.deviceId == null)) {
```
Comment (michalszorad) [inline apps/wallet-mobile/src/components/ConfirmTx/ConfirmTx.tsx:112]: Small refactoring

File: apps/wallet-mobile/src/components/ConfirmTx/ConfirmTx.tsx [hunk 2]
```diff
@@ -111,23 +119,36 @@ export const ConfirmTx = ({
...
 
-    if (chooseTransportOnConfirmation) {
-      await delay(1000)
-      onConfirm()
-    } else {
-      setDialogStep(DialogStep.Closed)
-    }
+    await wallet.ledgerSupportsCIP36(useUSB).then(async (isCIP36Supported) => {
+      if (supportsCIP36 !== isCIP36Supported) {
+        onCIP36SupportChange?.(isCIP36Supported)
```
Comment (michalszorad) [inline apps/wallet-mobile/src/components/ConfirmTx/ConfirmTx.tsx]: We need to re-generate transaction when we detect the wallet does not support CIP36

File: apps/wallet-mobile/src/components/ConfirmTx/ConfirmTx.tsx [hunk 3]
```diff
@@ -215,6 +237,11 @@ export const ConfirmTx = ({
     }
   }, [autoSignIfEasyConfirmation, wallet.isEasyConfirmationEnabled, _onConfirm])
 
+  useEffect(() => {
+    if (!autoConfirm || isProcessing || isProcessed) return
+    onConfirm()
```
Comment (michalszorad) [inline apps/wallet-mobile/src/components/ConfirmTx/ConfirmTx.tsx:245]: When TX is regenerated, we automatically confirm to continue in the flow

File: apps/wallet-mobile/package.json [hunk 4]
```diff
@@ -89,7 +89,7 @@
     "@emurgo/csl-mobile-bridge": "^5.1.2",
     "@emurgo/react-native-blockies-svg": "^0.0.2",
     "@emurgo/react-native-hid": "^5.15.6",
-    "@emurgo/yoroi-lib": "^0.5.2",
+    "@emurgo/yoroi-lib": "file:/Users/michalszorad/Documents/GitHub/yoroi-lib",
```
Comment (michalszorad) [inline apps/wallet-mobile/package.json]: This will need to be updated when https://github.com/Emurgo/yoroi-lib/pull/105 is resolved

**Review by banklesss — APPROVED** [review]

File: apps/wallet-mobile/src/Catalyst/ConfirmVotingTx.tsx [hunk 5]
```diff
@@ -25,15 +25,21 @@ export const ConfirmVotingTx = ({
   onNext: () => void
   pin: string
 }) => {
+  const [supportsCIP36, setSupportsCIP36] = useState<boolean>(true)
```
Comment (banklesss) [inline apps/wallet-mobile/src/Catalyst/ConfirmVotingTx.tsx]: dont needed right?

```suggestion
  const [supportsCIP36, setSupportsCIP36] = useState(true)
```

Comment (michalszorad) [reply apps/wallet-mobile/src/Catalyst/ConfirmVotingTx.tsx]: Correct

Comment (michalszorad) [reply apps/wallet-mobile/src/Catalyst/ConfirmVotingTx.tsx]: That's now removed

**Pull request merged**